sparse_strips/vello_cpu/src/render.rs

```diff
@@ -392,7 +415,9 @@ impl RenderContext {
     ) {
         // TODO: Maybe we should move those checks into the dispatcher.
         let wide = self.dispatcher.wide();
-        assert!(!wide.has_layers(), "some layers haven't been popped yet");
+        if self.ensure_all_layers_are_popped {
+            assert!(!wide.has_layers(), "some layers haven't been popped yet");
+        }
```

Comment:
I'd like there to be a test or two that this makes sense (i.e. performing some rendering with mismatched layer counts, then rendering again with the same context).

Because my assumption would be that this check has the potential for impact on rendering (for example, I was under the impression that certain blends would only happen on layer close).

Reply:
To expand on that - if this was always valid, I would probably lean towards making this a `debug_panic`, like we see in Masonry; i.e. panicking when debug assertions are enabled, and logging an error when not. That needn't happen in this PR, however.